File: backend/internal/logger/mobile_game_logger.go
package logger

import (
	"encoding/json"
	"fmt"
	"log"
	"os"
	"runtime"
	"time"

	"github.com/gin-gonic/gin"
)

// MobileGameLogger provides enhanced logging for mobile game development
type MobileGameLogger struct {
	*log.Logger
	startTime time.Time
	metrics   *PerformanceMetrics
}

type PerformanceMetrics struct {
	ActiveConnections    int           `json:"active_connections"`
	GameSessions         int           `json:"game_sessions"`
	AverageResponseTime  time.Duration `json:"average_response_time"`
	CardPlaysPerSecond   float64       `json:"card_plays_per_second"`
	DatabaseQueryTime    time.Duration `json:"database_query_time"`
	MemoryUsageMB        float64       `json:"memory_usage_mb"`
	GoroutineCount       int           `json:"goroutine_count"`
	LastHealthCheck      time.Time     `json:"last_health_check"`
}

type GameEvent struct {
	Timestamp   time.Time   `json:"timestamp"`
	PlayerID    string      `json:"player_id"`
	GameID      string      `json:"game_id"`
	EventType   string      `json:"event_type"`
	CardPlayed  *Card       `json:"card_played,omitempty"`
	TableState  []Card      `json:"table_state,omitempty"`
	IsValidPlay bool        `json:"is_valid_play"`
	Points      int         `json:"points"`
	RuleChecked string      `json:"rule_checked,omitempty"`
	ResponseTime time.Duration `json:"response_time"`
}

type Card struct {
	Value int    `json:"value"`
	Suit  string `json:"suit"`
}

type LogLevel string

const (
	DEBUG    LogLevel = "DEBUG"
	INFO     LogLevel = "INFO"
	SUCCESS  LogLevel = "SUCCESS"
	WARN     LogLevel = "WARN"
	ERROR    LogLevel = "ERROR"
	CRITICAL LogLevel = "CRITICAL"
)

var MobileLogger *MobileGameLogger

func init() {
	MobileLogger = NewMobileGameLogger()
}

func NewMobileGameLogger() *MobileGameLogger {
	logger := &MobileGameLogger{
		Logger:    log.New(os.Stdout, "", 0),
		startTime: time.Now(),
		metrics: &PerformanceMetrics{
			LastHealthCheck: time.Now(),
		},
	}

	logger.Success("MOBILE_LOGGER", "Mobile game development logger initialized", map[string]interface{}{
		"start_time": logger.startTime,
		"go_version": runtime.Version(),
		"goroutines": runtime.NumGoroutine(),
	})

	return logger
}

// ===== ROMANIAN SEPTICA GAME LOGGING =====

func (l *MobileGameLogger) LogCardPlay(playerID, gameID string, card Card, tableState []Card, isValid bool, points int, responseTime time.Duration) {
	event := GameEvent{
		Timestamp:   time.Now(),
		PlayerID:    playerID,
		GameID:      gameID,
		EventType:   "CARD_PLAY",
		CardPlayed:  &card,
		TableState:  tableState,
		IsValidPlay: isValid,
		Points:      points,
		ResponseTime: responseTime,
	}

	if isValid {
		l.Info("GAME_PLAY", fmt.Sprintf("Player %s played %d%s → %d points (%v)",
			playerID[:8], card.Value, card.Suit, points, responseTime), event)

		// Validate Romanian Septica rules
		l.validateRomanianRules(card, tableState, isValid, event)
	} else {
		l.Warn("INVALID_PLAY", fmt.Sprintf("Player %s invalid play: %d%s",
			playerID[:8], card.Value, card.Suit), event)
	}

	// Performance warning for slow card processing
	if responseTime > 200*time.Millisecond {
		l.Warn("PERFORMANCE", fmt.Sprintf("Slow card processing: %v", responseTime), map[string]interface{}{
			"player_id": playerID[:8],
			"response_time": responseTime,
			"target_time": "< 200ms",
		})
	}
}

func (l *MobileGameLogger) validateRomanianRules(card Card, tableState []Card, wasValid bool, event GameEvent) {
	// Rule: 7s always beat
	if card.Value == 7 {
		event.RuleChecked = "7_beats_all"
		if !wasValid {
			l.Error("RULE_VIOLATION", "7 should always beat but was rejected!", event)
		} else {
			l.Debug("RULE_CHECK", "7 beats all - valid", event)
		}
	}

	// Rule: Same values beat each other
	for _, tableCard := range tableState {
		if tableCard.Value == card.Value {
			event.RuleChecked = "same_value_beats"
			l.Debug("RULE_CHECK", fmt.Sprintf("%d beats %d - same value", card.Value, tableCard.Value), event)
			break
		}
	}

	// Rule: 8s beat when table cards count % 3 == 0
	if card.Value == 8 {
		if len(tableState)%3 == 0 {
			event.RuleChecked = "8_conditional_beat"
			l.Debug("RULE_CHECK", "8 beats (table count % 3 == 0)", event)
		} else {
			event.RuleChecked = "8_conditional_no_beat"
			l.Debug("RULE_CHECK", "8 doesn't beat (table count % 3 != 0)", event)
		}
	}

	// Log point cards (10s and Aces)
	if card.Value == 10 || card.Value == 14 { // Ace = 14
		pointType := "10 (1pt)"
		if card.Value == 14 {
			pointType = "Ace (1pt)"
		}
		l.Info("POINTS", fmt.Sprintf("Point card played: %s", pointType), map[string]interface{}{
			"card": fmt.Sprintf("%d%s", card.Value, card.Suit),
			"player_id": event.PlayerID[:8],
			"total_possible": 8,
		})
	}
}

func (l *MobileGameLogger) LogMatchmaking(playerID, status string, waitTime time.Duration) {
	data := map[string]interface{}{
		"player_id": playerID[:8],
		"status":    status,
		"wait_time": waitTime,
		"timestamp": time.Now(),
	}

	switch status {
	case "queued":
		l.Info("MATCHMAKING", fmt.Sprintf("Player %s joined queue", playerID[:8]), data)
	case "matched":
		l.Success("MATCHMAKING", fmt.Sprintf("Match found in %v", waitTime), data)
		l.updateMetric("matchmaking_success", 1)
	case "timeout":
		l.Warn("MATCHMAKING", fmt.Sprintf("Queue timeout after %v", waitTime), data)
	default:
		l.Info("MATCHMAKING", fmt.Sprintf("Status: %s", status), data)
	}
}

func (l *MobileGameLogger) LogGameSession(gameID string, event string, playerCount int) {
	data := map[string]interface{}{
		"game_id":      gameID,
		"event":        event,
		"player_count": playerCount,
		"timestamp":    time.Now(),
	}

	switch event {
	case "created":
		l.Success("GAME_SESSION", fmt.Sprintf("Game %s created with %d players", gameID[:8], playerCount), data)
		l.metrics.GameSessions++
	case "ended":
		l.Info("GAME_SESSION", fmt.Sprintf("Game %s ended", gameID[:8]), data)
		l.metrics.GameSessions--
	case "player_joined":
		l.Info("GAME_SESSION", fmt.Sprintf("Player joined game %s (%d total)", gameID[:8], playerCount), data)
	case "player_left":
		l.Warn("GAME_SESSION", fmt.Sprintf("Player left game %s (%d remaining)", gameID[:8], playerCount), data)
	}
}

// ===== WEBSOCKET/CONNECTION LOGGING =====

func (l *MobileGameLogger) LogWebSocketEvent(playerID, event string, latency time.Duration, error error) {
	data := map[string]interface{}{
		"player_id": playerID[:8],
		"event":     event,
		"latency":   latency,
		"timestamp": time.Now(),
	}

	if error != nil {
		data["error"] = error.Error()
	}

	switch event {
	case "connect":
		l.Success("WEBSOCKET", fmt.Sprintf("Player %s connected", playerID[:8]), data)
		l.metrics.ActiveConnections++
	case "disconnect":
		l.Info("WEBSOCKET", fmt.Sprintf("Player %s disconnected", playerID[:8]), data)
		l.metrics.ActiveConnections--
	case "reconnect":
		l.Info("WEBSOCKET", fmt.Sprintf("Player %s reconnected (%v)", playerID[:8], latency), data)
	case "message":
		if latency > 500*time.Millisecond {
			l.Warn("WEBSOCKET", fmt.Sprintf("High latency: %v", latency), data)
		} else {
			l.Debug("WEBSOCKET", fmt.Sprintf("Message processed (%v)", latency), data)
		}
	case "error":
		l.Error("WEBSOCKET", fmt.Sprintf("Connection error for %s", playerID[:8]), data)
	}
}

// ===== PERFORMANCE MONITORING =====

func (l *MobileGameLogger) UpdatePerformanceMetrics() {
	var m runtime.MemStats
	runtime.ReadMemStats(&m)

	l.metrics.MemoryUsageMB = float64(m.Alloc) / 1024 / 1024
	l.metrics.GoroutineCount = runtime.NumGoroutine()
	l.metrics.LastHealthCheck = time.Now()

	// Log performance warnings
	if l.metrics.MemoryUsageMB > 100 {
		l.Warn("PERFORMANCE", fmt.Sprintf("High memory usage: %.2f MB", l.metrics.MemoryUsageMB), l.metrics)
	}

	if l.metrics.GoroutineCount > 1000 {
		l.Warn("PERFORMANCE", fmt.Sprintf("High goroutine count: %d", l.metrics.GoroutineCount), l.metrics)
	}

	if l.metrics.ActiveConnections > 50 {
		l.Info("SCALE", fmt.Sprintf("High player count: %d active connections", l.metrics.ActiveConnections), l.metrics)
	}
}

func (l *MobileGameLogger) LogDatabaseQuery(query string, duration time.Duration, error error) {
	data := map[string]interface{}{
		"query":    query,
		"duration": duration,
		"timestamp": time.Now(),
	}

	l.metrics.DatabaseQueryTime = duration

	if error != nil {
		data["error"] = error.Error()
		l.Error("DATABASE", fmt.Sprintf("Query failed: %s", query), data)
	} else if duration > 100*time.Millisecond {
		l.Warn("DATABASE", fmt.Sprintf("Slow query (%v): %s", duration, query), data)
	} else {
		l.Debug("DATABASE", fmt.Sprintf("Query completed (%v)", duration), data)
	}
}

// ===== MOBILE GAME HEALTH MONITORING =====

func (l *MobileGameLogger) HealthCheck() map[string]interface{} {
	l.UpdatePerformanceMetrics()

	health := map[string]interface{}{
		"status":     "healthy",
		"uptime":     time.Since(l.startTime),
		"metrics":    l.metrics,
		"timestamp":  time.Now(),
		"issues":     []string{},
	}

	var issues []string

	// Check performance thresholds
	if l.metrics.MemoryUsageMB > 200 {
		issues = append(issues, fmt.Sprintf("High memory: %.2f MB", l.metrics.MemoryUsageMB))
	}

	if l.metrics.AverageResponseTime > 200*time.Millisecond {
		issues = append(issues, fmt.Sprintf("High response time: %v", l.metrics.AverageResponseTime))
	}

	if l.metrics.DatabaseQueryTime > 100*time.Millisecond {
		issues = append(issues, fmt.Sprintf("Slow database: %v", l.metrics.DatabaseQueryTime))
	}

	if len(issues) > 0 {
		health["status"] = "warning"
		health["issues"] = issues
		l.Warn("HEALTH_CHECK", fmt.Sprintf("Performance issues detected: %d", len(issues)), health)
	} else {
		l.Success("HEALTH_CHECK", "All systems optimal", health)
	}

	return health
}

// ===== STRUCTURED LOGGING METHODS =====

func (l *MobileGameLogger) Debug(category, message string, data interface{}) {
	l.log(DEBUG, category, message, data)
}

func (l *MobileGameLogger) Info(category, message string, data interface{}) {
	l.log(INFO, category, message, data)
}

func (l *MobileGameLogger) Success(category, message string, data interface{}) {
	l.log(SUCCESS, category, message, data)
}

func (l *MobileGameLogger) Warn(category, message string, data interface{}) {
	l.log(WARN, category, message, data)
}

func (l *MobileGameLogger) Error(category, message string, data interface{}) {
	l.log(ERROR, category, message, data)
}

func (l *MobileGameLogger) Critical(category, message string, data interface{}) {
	l.log(CRITICAL, category, message, data)
}

func (l *MobileGameLogger) log(level LogLevel, category, message string, data interface{}) {
	timestamp := time.Now().Format("15:04:05.000")

	logEntry := map[string]interface{}{
		"timestamp": timestamp,
		"level":     level,
		"category":  category,
		"message":   message,
		"uptime":    time.Since(l.startTime).Truncate(time.Millisecond),
	}

	if data != nil {
		logEntry["data"] = data
	}

	// Color coding for console output
	var color string
	switch level {
	case DEBUG:
		color = "\033[90m" // Gray
	case INFO:
		color = "\033[94m" // Blue
	case SUCCESS:
		color = "\033[92m" // Green
	case WARN:
		color = "\033[93m" // Yellow
	case ERROR:
		color = "\033[91m" // Red
	case CRITICAL:
		color = "\033[91;1m" // Bold Red
	}
	reset := "\033[0m"

	// Format for mobile game development
	if data != nil {
		jsonData, _ := json.Marshal(data)
		fmt.Printf("%s[%s] [%s] [%s] %s %s%s\n",
			color, timestamp, level, category, message, string(jsonData), reset)
	} else {
		fmt.Printf("%s[%s] [%s] [%s] %s%s\n",
			color, timestamp, level, category, message, reset)
	}
}

func (l *MobileGameLogger) updateMetric(metric string, value interface{}) {
	// Simple metric updates - in production would use proper metrics store
	switch metric {
	case "matchmaking_success":
		// Track successful matchmaking
	case "card_play_response":
		// Track card play response times
	}
}

// ===== GIN MIDDLEWARE FOR REQUEST LOGGING =====

func (l *MobileGameLogger) GinMiddleware() gin.HandlerFunc {
	return func(c *gin.Context) {
		start := time.Now()
		path := c.Request.URL.Path
		raw := c.Request.URL.RawQuery

		// Process request
		c.Next()

		// Calculate response time
		latency := time.Since(start)

		// Update performance metrics
		l.metrics.AverageResponseTime = latency

		if raw != "" {
			path = path + "?" + raw
		}

		data := map[string]interface{}{
			"method":     c.Request.Method,
			"path":       path,
			"status":     c.Writer.Status(),
			"latency":    latency,
			"ip":         c.ClientIP(),
			"user_agent": c.Request.UserAgent(),
		}

		if latency > 1*time.Second {
			l.Error("HTTP_SLOW", fmt.Sprintf("%s %s took %v", c.Request.Method, path, latency), data)
		} else if c.Writer.Status() >= 400 {
			l.Warn("HTTP_ERROR", fmt.Sprintf("%s %s returned %d", c.Request.Method, path, c.Writer.Status()), data)
		} else {
			l.Debug("HTTP", fmt.Sprintf("%s %s (%v)", c.Request.Method, path, latency), data)
		}
	}
}

File: frontend/js/mobile-game-logger.js
// Mobile Game Development Logger for Romanian Septica
// Optimized for real-time debugging and performance monitoring

class MobileGameLogger {
    constructor() {
        this.startTime = performance.now();
        this.frameCount = 0;
        this.lastFpsTime = this.startTime;
        this.memoryBaseline = this.getMemoryUsage();
        this.touchEvents = [];
        this.gameEvents = [];

        // Performance monitoring
        this.performanceMetrics = {
            fps: 0,
            memory: 0,
            touchLatency: 0,
            wsLatency: 0,
            cardPlayTime: 0,
            renderTime: 0
        };

        // Start performance monitoring
        this.startPerformanceMonitoring();

        console.log('%c🎮 MOBILE GAME LOGGER INITIALIZED',
            'color: #4CAF50; font-weight: bold; font-size: 14px;');
    }

    // ===== PERFORMANCE MONITORING =====

    startPerformanceMonitoring() {
        // FPS Monitoring
        const fpsCounter = () => {
            this.frameCount++;
            const now = performance.now();
            const delta = now - this.lastFpsTime;

            if (delta >= 1000) {
                this.performanceMetrics.fps = Math.round((this.frameCount * 1000) / delta);
                this.frameCount = 0;
                this.lastFpsTime = now;

                // Log FPS if below mobile game targets
                if (this.performanceMetrics.fps < 30) {
                    this.warn('PERFORMANCE', `Low FPS: ${this.performanceMetrics.fps} (Target: 60)`, {
                        fps: this.performanceMetrics.fps,
                        memory: this.getMemoryUsage(),
                        timestamp: this.getTimestamp()
                    });
                }
            }

            requestAnimationFrame(fpsCounter);
        };
        requestAnimationFrame(fpsCounter);

        // Memory monitoring every 5 seconds
        setInterval(() => {
            const currentMemory = this.getMemoryUsage();
            this.performanceMetrics.memory = currentMemory;

            if (currentMemory > this.memoryBaseline * 2) {
                this.error('MEMORY', `Memory leak detected: ${currentMemory}MB (baseline: ${this.memoryBaseline}MB)`, {
                    current: currentMemory,
                    baseline: this.memoryBaseline,
                    ratio: (currentMemory / this.memoryBaseline).toFixed(2)
                });
            }
        }, 5000);
    }

    getMemoryUsage() {
        if (performance.memory) {
            return Math.round(performance.memory.usedJSHeapSize / 1024 / 1024);
        }
        return 0;
    }

    // ===== ROMANIAN SEPTICA GAME LOGGING =====

    logCardPlay(playerName, card, tableState, isValidPlay, points = 0) {
        const timestamp = this.getTimestamp();
        const logData = {
            player: playerName,
            card: `${card.value}${card.suit}`,
            tableState: tableState.map(c => `${c.value}${c.suit}`),
            valid: isValidPlay,
            points: points,
            timestamp: timestamp,
            gameTime: this.getGameTime()
        };

        if (isValidPlay) {
            this.info('GAME_PLAY', `${playerName} played ${card.value}${card.suit} → ${points} points`, logData);
        } else {
            this.warn('GAME_RULE', `Invalid play: ${playerName} tried ${card.value}${card.suit}`, logData);
        }

        // Check Romanian Septica rules
        this.validateRomanianRules(card, tableState, isValidPlay);
    }

    validateRomanianRules(playedCard, tableState, wasValid) {
        const rules = {
            '7_beats_all': playedCard.value === 7,
            'same_value_beats': tableState.some(c => c.value === playedCard.value),
            '8_conditional': playedCard.value === 8 && (tableState.length % 3 === 0),
            'point_card': playedCard.value === 10 || playedCard.value === 14 // Ace = 14
        };

        // Log rule violations
        if (!wasValid && rules['7_beats_all']) {
            this.error('RULE_VIOLATION', '7 should always beat but was rejected!', {
                card: `${playedCard.value}${playedCard.suit}`,
                rule: '7_beats_all',
                tableState: tableState.map(c => `${c.value}${c.suit}`)
            });
        }

        if (rules.point_card) {
            this.info('POINTS', `Point card played: ${playedCard.value}${playedCard.suit}`, {
                value: playedCard.value === 10 ? '10 (1pt)' : 'Ace (1pt)',
                totalPossible: 8
            });
        }
    }

    logMatchmaking(status, playerId, waitTime = 0) {
        const logData = {
            playerId: playerId,
            status: status,
            waitTime: waitTime,
            timestamp: this.getTimestamp()
        };

        switch (status) {
            case 'queued':
                this.info('MATCHMAKING', `Player ${playerId} joined queue`, logData);
                break;
            case 'matched':
                this.success('MATCHMAKING', `Match found in ${waitTime}ms`, logData);
                break;
            case 'timeout':
                this.warn('MATCHMAKING', `Queue timeout after ${waitTime}ms`, logData);
                break;
            default:
                this.info('MATCHMAKING', `Status: ${status}`, logData);
        }
    }

    // ===== WEBSOCKET/NETWORK LOGGING =====

    logWebSocketEvent(eventType, data = null, latency = 0) {
        const timestamp = this.getTimestamp();
        const logData = {
            event: eventType,
            latency: latency,
            timestamp: timestamp,
            data: data
        };

        switch (eventType) {
            case 'connect':
                this.success('WEBSOCKET', 'Connected to game server', logData);
                break;
            case 'disconnect':
                this.error('WEBSOCKET', 'Disconnected from server', logData);
                break;
            case 'reconnect':
                this.info('WEBSOCKET', `Reconnected (${latency}ms)`, logData);
                break;
            case 'message':
                if (latency > 500) {
                    this.warn('WEBSOCKET', `High latency: ${latency}ms`, logData);
                } else {
                    this.debug('WEBSOCKET', `Message received (${latency}ms)`, logData);
                }
                break;
            case 'error':
                this.error('WEBSOCKET', 'Connection error', logData);
                break;
        }

        this.performanceMetrics.wsLatency = latency;
    }

    // ===== TOUCH/INPUT LOGGING =====

    logTouchEvent(eventType, touchData, responseTime = 0) {
        const timestamp = this.getTimestamp();
        const logData = {
            type: eventType,
            x: touchData.x,
            y: touchData.y,
            responseTime: responseTime,
            timestamp: timestamp
        };

        this.touchEvents.push(logData);

        // Keep only last 10 touch events
        if (this.touchEvents.length > 10) {
            this.touchEvents.shift();
        }

        if (responseTime > 100) {
            this.warn('INPUT_LAG', `Slow touch response: ${responseTime}ms`, logData);
        } else {
            this.debug('TOUCH', `${eventType} (${responseTime}ms)`, logData);
        }

        this.performanceMetrics.touchLatency = responseTime;
    }

    // ===== MOBILE DEVICE SPECIFIC =====

    logDeviceInfo() {
        const deviceInfo = {
            userAgent: navigator.userAgent,
            devicePixelRatio: window.devicePixelRatio,
            screenSize: `${screen.width}x${screen.height}`,
            viewportSize: `${window.innerWidth}x${window.innerHeight}`,
            connection: navigator.connection ? navigator.connection.effectiveType : 'unknown',
            memory: navigator.deviceMemory || 'unknown',
            cores: navigator.hardwareConcurrency || 'unknown',
            online: navigator.onLine,
            webgl: this.checkWebGLSupport()
        };

        this.info('DEVICE', 'Mobile device capabilities detected', deviceInfo);
        return deviceInfo;
    }

    checkWebGLSupport() {
        const canvas = document.createElement('canvas');
        const gl = canvas.getContext('webgl2') || canvas.getContext('webgl');
        return gl ? 'WebGL2/WebGL' : 'No WebGL';
    }

    // ===== ERROR HANDLING =====

    logError(error, context = {}) {
        const errorData = {
            message: error.message,
            stack: error.stack,
            context: context,
            timestamp: this.getTimestamp(),
            gameTime: this.getGameTime(),
            performance: this.performanceMetrics
        };

        this.error('GAME_ERROR', error.message, errorData);

        // Auto-report critical errors
        if (error.name === 'TypeError' || error.name === 'ReferenceError') {
            this.error('CRITICAL', 'Game-breaking error detected', errorData);
        }
    }

    // ===== LOG FORMATTING =====

    info(category, message, data = null) {
        this.log('INFO', category, message, data, 'color: #2196F3');
    }

    success(category, message, data = null) {
        this.log('SUCCESS', category, message, data, 'color: #4CAF50; font-weight: bold');
    }

    warn(category, message, data = null) {
        this.log('WARN', category, message, data, 'color: #FF9800; font-weight: bold');
    }

    error(category, message, data = null) {
        this.log('ERROR', category, message, data, 'color: #F44336; font-weight: bold');
    }

    debug(category, message, data = null) {
        if (this.debugMode) {
            this.log('DEBUG', category, message, data, 'color: #9E9E9E');
        }
    }

    log(level, category, message, data = null, style = '') {
        const timestamp = this.getTimestamp();
        const prefix = `[${timestamp}] [${level}] [${category}]`;

        if (data) {
            console.log(`%c${prefix} ${message}`, style, data);
        } else {
            console.log(`%c${prefix} ${message}`, style);
        }
    }

    // ===== UTILITY METHODS =====

    getTimestamp() {
        return new Date().toLocaleTimeString('en-US', {
            hour12: false,
            hour: '2-digit',
            minute: '2-digit',
            second: '2-digit',
            fractionalSecondDigits: 3
        });
    }

    getGameTime() {
        return Math.round(performance.now() - this.startTime);
    }

    // ===== PERFORMANCE DASHBOARD =====

    showPerformanceStats() {
        const stats = {
            ...this.performanceMetrics,
            gameTime: this.getGameTime(),
            memory: this.getMemoryUsage(),
            touchEvents: this.touchEvents.length
        };

        console.table(stats);
        return stats;
    }

    // ===== MOBILE GAME SPECIFIC ALERTS =====

    checkMobileGameHealth() {
        const issues = [];

        if (this.performanceMetrics.fps < 30) {
            issues.push(`Low FPS: ${this.performanceMetrics.fps}`);
        }

        if (this.performanceMetrics.memory > 100) {
            issues.push(`High memory: ${this.performanceMetrics.memory}MB`);
        }

        if (this.performanceMetrics.touchLatency > 100) {
            issues.push(`Input lag: ${this.performanceMetrics.touchLatency}ms`);
        }

        if (this.performanceMetrics.wsLatency > 300) {
            issues.push(`Network lag: ${this.performanceMetrics.wsLatency}ms`);
        }

        if (issues.length > 0) {
            this.warn('HEALTH_CHECK', 'Mobile game performance issues detected', {
                issues: issues,
                recommendations: this.getPerformanceRecommendations()
            });
        } else {
            this.success('HEALTH_CHECK', 'Mobile game performance is optimal', this.performanceMetrics);
        }

        return issues;
    }

    getPerformanceRecommendations() {
        const recommendations = [];

        if (this.performanceMetrics.fps < 30) {
            recommendations.push('Reduce Three.js quality settings');
            recommendations.push('Enable mobile LOD system');
        }

        if (this.performanceMetrics.memory > 100) {
            recommendations.push('Check for memory leaks in card animations');
            recommendations.push('Dispose unused Three.js objects');
        }

        return recommendations;
    }
}

// Global mobile game logger instance
window.MobileGameLogger = new MobileGameLogger();

// Auto-detect mobile device capabilities on load
window.addEventListener('load', () => {
    window.MobileGameLogger.logDeviceInfo();

    // Set up error tracking
    window.addEventListener('error', (event) => {
        window.MobileGameLogger.logError(event.error, {
            filename: event.filename,
            lineno: event.lineno,
            colno: event.colno
        });
    });

    // Monitor for mobile-specific events
    if ('ontouchstart' in window) {
        document.addEventListener('touchstart', (e) => {
            const startTime = performance.now();
            requestAnimationFrame(() => {
                const responseTime = performance.now() - startTime;
                window.MobileGameLogger.logTouchEvent('touchstart', {
                    x: e.touches[0].clientX,
                    y: e.touches[0].clientY
                }, responseTime);
            });
        });
    }

    // Health check every 30 seconds
    setInterval(() => {
        window.MobileGameLogger.checkMobileGameHealth();
    }, 30000);
});

export default MobileGameLogger;